.changeset/chilly-pants-fix.md

```diff
@@ -0,0 +1,24 @@
+---
+'astro': minor
+---
+
+Astro exposes the middleware file path to the integrations in the hook `astro:build:done`
+
+```ts
+// myIntegration.js
+import type { AstroIntegration } from 'astro';
+function integration(): AstroIntegration {
+    return {
+        name: "fancy-astro-integration",
+        hooks: {
+            'astro:build:done': ({ middlewarePath }) => { 
+                if (middlewarePath) {
+                    // do some operations
+                }
+            }
```

Comment:
Wonder if it makes sense to name this `middlewareEntryPoint` instead? So it's consistent with `entryPoints`